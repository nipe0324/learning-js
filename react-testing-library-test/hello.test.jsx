import React from 'react'
import '@testing-library/jest-dom';
import { render, screen } from "@testing-library/react";

import Hello from "./hello";

describe('Hello', () => {
  it('renders "Hey, stranger"', () => {
    render(<Hello />);
    expect(screen.getByText('Hey, stranger')).toBeInTheDocument();
  });

  it('renders "Hello, John" with the name props', () => {
    render(<Hello name="John" />);
    expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent('Hello, John');
  });

  it('renders Hello component', () => {
    const { asFragment } = render(<Hello />);

    expect(asFragment()).toMatchSnapshot();
  })
});
