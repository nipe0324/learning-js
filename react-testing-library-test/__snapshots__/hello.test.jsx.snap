// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello renders Hello component 1`] = `
<DocumentFragment>
  <span>
    Hey, stranger
  </span>
</DocumentFragment>
`;
